.github/workflows/library_dafny_verification.yml

```diff
@@ -48,8 +59,18 @@ jobs:
         with:
           dotnet-version: "6.0.x"
 
+      - name: Regenerate code using smithy-dafny
+        working-directory: ./${{ matrix.library }}
+        run: |
+          make polymorph_dafny
+
+      - name: Check regenerated code against commited code
+        if: inputs.diff-generated-code
+        working-directory: ./${{ matrix.library }}
+        run: |
+          make check_polymorph_diff
```

Comment:
I also wonder about this. Is this scope creep for the code base?
Would it be better to do the code-gen in a single CI step that rather than for many different OSs/places/times etc?

Reply:
The challenge is that for the nightly build (or any case of building for a different Dafny version) you need to regenerate the code for it to work at all. So the "Regenerate code using smithy-dafny" step has to be done before the rest, to leave the generated code in the right state.

We could only have the diff check in a separate CI step so it didn't have to be conditional, but that would just be even more expensive than this solution.

Reply:
When you say that the diff check in a separate CI step is "more expensive" what do you mean?